src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,106 @@
+#include <hyro/SignalGeneratorComponent.h>
+
+using namespace hyro;
+
+namespace signal_generator
+{
+
+std::shared_ptr<hyro::HyroLogger> SignalGeneratorComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+
+hyro::Result
+SignalGeneratorComponent::init(const hyro::ComponentConfiguration &config)
+{
+  m_amplitude = 15.0;
+  m_frequency = 30.0;
+  m_cosine = true;
+  m_name = "Name 1";
+
+  m_signal.setAmp(m_amplitude);
+  m_signal.setDeg(m_frequency);
+  m_signal.setStep(5);
+
+  signal_output = this->registerOutput<hyro::SignalMsgs>("analog_output"_uri, config);
+
+  std::shared_ptr<ChannelOutput<std::vector<int>>>
+  m_dummy = this->registerOutput<std::vector<int>>("fix_dynamic"_uri, config);
+
+  registerDynamicProperty<float>("Amplitude",
```

Comment:
Usually, we use **snake_case** style in the dynamic parameter and channels names.